files/assets/js/comments.js

```diff
@@ -57,3 +57,49 @@ function expandMarkdown(t,id) {
 	if (val.innerHTML == 'View source') val.innerHTML = 'Hide source'
 	else val.innerHTML = 'View source'
 };
+
+function findDepth(element) {
+	var parents = [];
+	while(element.parentNode && element.parentNode.nodeName.toLowerCase() != 'body') {
+		element = element.parentNode;
+		if(element.classList.contains("comment")){
+			parents.push(element);
+		}
+	}
+	return parents.length;
+}
+
+function colorize(element) {
+
+	function getColor(depth) {
+		while(depth > 9) depth -= 9;
+		return "collapse-bar-" + depth;
+	}
+
+	const classname = "comment-collapse-bar";
+	let bars = [];
```

Comment:
This section seems a bit confusing. Thinking some about ways it might be refactored - what if the section inside the for loop below was a function instead, then both cases inside the first if block could call it once, and the loop could be inside the first else block to call that function for each element?